File: pi_MonteCarlo.cpp
#include <stdio.h>  //라이브러리 : 이미 만들어진 함수 모음 
#include <stdlib.h>  //랜덤 함수 라이브러리 
#include <time.h>  //시간 함수 라이브러리

int main(){
	double pi,x,y,len=0,cr_size,sq_size;
	double max=32767;
		
	printf("반복 수 입력 : ");
	scanf("%lf",&sq_size);  //반복할 횟수 입력 
		
	srand(time(NULL));  //난수 초기화 
	
	for(int i=0;i<sq_size;i++){  //입력한 수 만큼 반복 
		x=rand()/max;  //랜덤 x 좌표 
		y=rand()/max;  //랜덤 y 좌표 
		len=(x*x) + (y*y);  //원점으로 부터의 거리 
		
		if(len<=1){  //만약 거리가 1보다 작거나 같다면 
			cr_size++;  //원넓이 1 추가 
		}
		pi=4*(cr_size/sq_size);  //원주율 구하기 반*반*pi*1/4=cr_size 
	}
	printf("pi : %.15lf\n",pi);  //결과값 출력
	
	printf("\nPress enter to exit");
	system("pause>nul");
	return 0;
}	

File: pi_MonteCarlo.py
import random  # 난수 생성 함수 라이브러리

sq_size = int(input("반복 수 입력 : "))  # 찍을 점의 개수 입력
cr_size = 0  # 원의 넓이 저장하는 변수 선언

for i in range(sq_size):  # 입력한 만큼 반복
    x = random.random()  # 랜덤 x좌표 생성
    y = random.random()  # 랜덤 y좌표 생성
    len = (x*x) + (y*y)  # 원점으로부터의 거리 계산

    if(len <= 1):  # 만약 거리가 1보다 작거나 같으면
        cr_size += 1  # 원의 넓이에 1만큼 더하라

pi = 4*(cr_size/sq_size)  # 원주율 계산
print("pi : %.15lf" % pi)  # 결괏값 출력